### PR TITLE
feat: reduce package permissions in maven builder / publisher

### PR DESCRIPTION
**Pull request by AdamKorcz**
Write permissions are not needed for the lowperms delegator, so removing that from the Maven builder.

### CODE REVIEW

File: internal/builders/maven/README.md [hunk 1]
```diff
@@ -84,7 +84,6 @@ jobs:
       contents: write
```
Comment (laurentsimon) [inline internal/builders/maven/README.md:84]: is contents:write needed?

File: internal/builders/maven/README.md [hunk 2]
```diff
@@ -81,11 +81,10 @@ env:
 jobs:
   build:
     permissions:
```
Comment (laurentsimon) [inline internal/builders/maven/README.md:83]: the publisher call still uses write permissions:
```yaml
permissions:
      contents: write
      id-token: write
      actions: read
      packages: write
```, 

None of them are needed, correct?

**Review by ianlewis — COMMENTED** [review]
This LGTM but we should address @laurentsimon's comments.

File: internal/builders/maven/README.md (hunk 2 above)
Comment (ianlewis) [reply internal/builders/maven/README.md:83]: > None of them are needed, correct?

What do you mean here? IIUC we need these permissions after `packages: write` is removed from the call to the lowperms delegator.

```yaml
permissions:
  id-token: write
  contents: read
  actions: read
```

Comment (laurentsimon) [reply internal/builders/maven/README.md:83]: sorry, that was meat as a comment for the publish, not the build. The publish should not need permissions

File: .github/workflows/publish_maven.yml [hunk 3]
```diff
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+permissions: read-all
```
Comment (laurentsimon) [inline .github/workflows/publish_maven.yml:15]: added this

File: internal/builders/maven/README.md [hunk 4]
```diff
@@ -97,12 +96,7 @@ You can also release artifacts to Maven Central by adding the following step to
 ```yaml
   publish:
     needs: build
-    permissions:
```
Comment (laurentsimon) [inline internal/builders/maven/README.md:100]: I removed the permissions.

File: internal/builders/maven/README.md [hunk 5]
```diff
@@ -81,11 +81,10 @@ env:
 jobs:
   build:
     permissions:
-      contents: write
       id-token: write
+      contents: read
       actions: read
-      packages: write
-    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_maven_slsa3.yml@main
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_maven_slsa3.yml@v1.7.0
```
Comment (laurentsimon) [inline internal/builders/maven/README.md:87]: I updated the ref

**Review by laurentsimon — APPROVED** [review]

**Pull request merged**